internal/go_repository.bzl

```diff
@@ -147,7 +147,13 @@ def _go_repository_impl(ctx):
             print("fetch_repo: " + result.stderr)
 
     if generate:
-        # Build file generation is needed
+        # Build file generation is needed. Populate Gazelle directive at root build file
+        ctx.file(
+            ctx.attr.build_file_name.split(",")[0],
+            "\n".join(["# gazelle:" + d for d in ctx.attr.gazelle_directives]),
```

Comment:
Let's call this attribute `directives`, and let's not force elements to start with `gazelle:`. I think buildifier has some of its own directives, and Gazelle integrates part of that. It's possible other tools with different prefixes may be used in the future.